pkg/restapi/operation/common/models.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package common
+
+import (
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
+)
+
+// CreatePublicDIDRequest model
+//
+// This is used for operation to create public DID
+//
+// swagger:parameters createPublicDID
+type CreatePublicDIDRequest struct {
+	// Params for creating public DID
+	//
+	// in: path
+	*CreatePublicDIDParams
+}
+
+// CreatePublicDIDParams contains parameters for creating new public DID
+type CreatePublicDIDParams struct {
+	// Params for creating public DID
+	Method string `json:"method"`
+
+	// RequestHeader to be included while submitting request to http binding URL
+	RequestHeader string `json:"header"`
+}
+
+// CreatePublicDIDResponse model
+//
+// This is used for returning public DID created
+//
+// swagger:response createPublicDIDResponse
+type CreatePublicDIDResponse struct {
+
+	// in: body
+	DID *did.Doc `json:"did"`
```

Comment:
I believe it's better to return the base64-encoded raw bytes of the DID doc for the reasons laid out in hyperledger/aries-rfcs#243

Reply:
Agreed, will be addressed in #855 , added a TODO